src/rules.jl

```diff
@@ -58,10 +58,25 @@ will be hit as a fallback. This is the case for most rules.
 
 See also: [`rrule`](@ref), [`@scalar_rule`](@ref), [`RuleConfig`](@ref)
 """
-frule(::Any, ::Any, ::Vararg{Any}; kwargs...) = nothing
+frule(::Any, ::Any, ::Vararg{Any}) = nothing
 
 # if no config is present then fallback to config-less rules
-frule(::RuleConfig, ȧrgs, f, args...; kwargs...) = frule(ȧrgs, f, args...; kwargs...)
+frule(::RuleConfig, args...) = frule(args...)
+
+# Manual fallback for keyword arguments. Usually this would be generated by
+#
+#   frule(::Any, ::Vararg{Any}; kwargs...) = nothing
+#
+# However - the fallback method is so hot that we want to avoid any extra code
+# that would be required to have the automatically generated method package up
+# the keyword arguments (which the optimizer will throw away, but the compiler
+# still has to manually analyze). Manually declare this method with an
+# explicitly empty body to save the compiler that work.
+const frule_kwfunc = Core.kwftype(typeof(frule)).instance
+(::typeof(frule_kwfunc))(::Any, ::typeof(frule), ::Any, ::Vararg{Any}) = nothing
```

Comment:
I guess either having the two `Any`s
```suggestion
(::typeof(frule_kwfunc))(::Any, ::typeof(frule), ::Any, ::Any, ::Vararg{Any}) = nothing
```
or pack them all into `Vararg`
```suggestion
(::typeof(frule_kwfunc))(::Any, ::typeof(frule), ::Vararg{Any}) = nothing
```
right?

Reply:
Yeah, let me make it consistent.